ecosystem/typescript/sdk_v2/src/core/hex.ts

```diff
@@ -0,0 +1,120 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+import { bytesToHex, hexToBytes } from "@noble/hashes/utils";
+import { HexInput } from "../types";
+import { ParsingError, ParsingResult } from "./common";
+
+/**
+ * This enum is used to explain why parsing might have failed.
+ */
+export enum HexInvalidReason {
+  TOO_SHORT = "too_short",
+  INVALID_LENGTH = "invalid_length",
+  INVALID_HEX_CHARS = "invalid_hex_chars",
+}
+
+/**
+ * NOTE: Do not use this class when working with account addresses, use AccountAddress.
+ *
+ * NOTE: When accepting hex data as input to a function, prefer to accept HexInput and
+ * then use the static helper methods of this class to convert it into the desired
+ * format. This enables the greatest flexibility for the developer.
+ *
+ * Hex is a helper class for working with hex data. Hex data, when represented as a
+ * string, generally looks like this, for example: 0xaabbcc, 45cd32, etc.
+ *
+ * You might use this class like this:
+ *
+ * ```ts
+ * getTransactionByHash(txnHash: HexInput): Promise<Transaction> {
+ *   const txnHashString = Hex.hexInputToString({ hexInput: txnHash });
+ *   return await getTransactionByHashInner(txnHashString);
+ * }
+ * ```
+ *
+ * This call to `Hex.hexInputToString` converts the HexInput to a hex string with a
+ * leading 0x prefix, regardless of what the input format was.
+ */
+export class Hex {
+  private data: Uint8Array;
+
+  constructor(args: { data: Uint8Array }) {
+    this.data = args.data;
+  }
+
+  static fromString(args: { str: string }): Hex {
```

Comment:
nit: can reorganize this file where all non-static methods at the top and follow by static methods